### PR TITLE
Contingency by Filters

### PR DESCRIPTION
**Pull request by ChamseddineBhd**

### CODE REVIEW

**Review by geofjamg — CHANGES_REQUESTED** [review]
GUI to remove everywhere

File: src/components/filters-editor.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                    <FormControl className="MuiFormControl-root">
+                        <NativeSelect
+                            id="demo-customized-select-native"
+                            value={equipmentType}
+                            onChange={handleEquipmentType}
+                            input={<BootstrapInput />}
+                        >
+                            <option value={'*'}>*</option>
+                            <option value={'Lines'}>Lines</option>
+                            <option value={'Substations'}>Substations</option>
```
Comment (med-hedi) [inline src/components/filters-editor.js:130]: You need to translate all labels and options text in this file

File: src/components/filters-editor.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                            <option value={'>='}> &ge; </option>
+                            <option value={'<'}> &lt; </option>
+                            <option value={'<='}> &le; </option>
+                        </Select>
+                    </FormControl>
+                </Grid>
+                <Grid item xs={12} sm={8}>
+                    <TextField
+                        onChange={handleNominalVoltage}
+                        label="Nominal voltage"
```
Comment (med-hedi) [inline src/components/filters-editor.js]: You need to translate all labels and options text in this file

**Comment by med-hedi** [comment]
In light mode the list of options in the equipment type selectbox is not visible
selectBox

**Comment by med-hedi** [comment]
When adding a new list, the cancel and save buttons should be disabled if we have not yet inserted any code in AceEditor

**Comment by ChamseddineBhd** [comment]
> When adding a new list, the cancel and save buttons should be disabled if we have not yet inserted any code in AceEditor

When we create a new List by using filters we can save the list even with no changes, for consistency i think we should accept to save an empty script as we can save a script with space for example!

**Review by FranckLecuyer — APPROVED** [review]

File: src/components/filters-editor.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+import { FormattedMessage, useIntl } from 'react-intl';
+
+const useStyles = makeStyles(() => ({
+    root: {
+        flexGrow: 1,
+        paddingLeft: '40px',
+    },
+    textField: {
+        color: 'red',
+        width: '2033px',
```
Comment (med-hedi) [inline src/components/filters-editor.js]: Why 2033px ?

**Review by geofjamg — CHANGES_REQUESTED** [review]
A script list is selected but the script content is not displayed:
image

File: src/components/filters-editor.js (hunk 3 above)
Comment (geofjamg) [reply src/components/filters-editor.js]: @ChamseddineBhd, @med-hedi is right, this is not a good practice to hardcode this. Is the app still "not so bad" responsive?

File: src/translations/fr.json [hunk 4]
```diff
@@ -18,5 +18,18 @@
...
+    "equipmentType": "Type d'équipement",
+    "equipmentName": "Nom d'équipement",
+    "equipmentID": "ID d'équipement",
+    "nominalVoltage": "Tension nominale",
+    "SCRIPT": "Script",
+    "FILTERS": "Filtres",
+    "LINE": "LIGNE",
+    "TWO_WINDINGS_TRANSFORMER": "TRANSFO 2E",
+    "GENERATOR": "GÉNÉRATEUR",
+    "STATIC_VAR_COMPENSATOR": "SVC",
```
Comment (geofjamg) [inline src/translations/fr.json]: CSPR

File: src/translations/en.json [hunk 5]
```diff
@@ -18,5 +18,18 @@
     "rename": "Rename",
     "delete": "Delete",
     "renameList": "Rename list",
-    "newNameList": "New list name"
+    "newNameList": "New list name",
+    "equipmentType": "Equipment type",
+    "equipmentName": "Equipment Name",
+    "equipmentID": "Equipment ID",
+    "nominalVoltage": "nominal Voltage",
```
Comment (geofjamg) [inline src/translations/en.json]: Upper case

File: src/translations/en.json [hunk 6]
```diff
@@ -18,5 +18,18 @@
...
+    "SCRIPT": "Script",
+    "FILTERS": "Filters",
+    "TWO_WINDINGS_TRANSFORMER": "2W TRANSFORMER",
```
Comment (geofjamg) [inline src/translations/en.json:28]: Why do you put translation in upper case?

File: src/translations/fr.json [hunk 7]
```diff
@@ -18,5 +18,18 @@
     "rename": "Renommer",
     "delete": "Supprimer",
     "renameList": "Renommer la liste",
-    "newNameList": "Nouveau nom de la liste"
+    "newNameList": "Nouveau nom de la liste",
+    "equipmentType": "Type d'équipement",
```
Comment (geofjamg) [inline src/translations/fr.json:22]: Why do you put translation in upper case?

**Review by geofjamg — CHANGES_REQUESTED** [review]

**Comment by geofjamg** [comment]
> A script list is selected but the script content is not displayed:
> image

My bad, it was a deployment error on my side

**Review by geofjamg — APPROVED** [review]

**Pull request merged**